ivy/array/array.py

```diff
@@ -298,10 +298,11 @@ def __repr__(self):
         )
         arr_np = backend.to_numpy(self._data)
         rep = ivy.vec_sig_fig(arr_np, sig_fig) if self._size > 0 else np.array(arr_np)
+        repr = rep.__repr__()[:-1].partition(", dtype")[0].partition(", dev")[0]
```

Comment:
probably best to keep this line inside the `with` statement to get the correct number of decimals, otherwise lgtm!

Reply:
Got it thanks!